src/LEGO.AsyncAPI/Writers/AsyncApiYamlWriter.cs

```diff
@@ -1,25 +1,48 @@
 ﻿// Copyright (c) The LEGO Group. All rights reserved.
 
+using System;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
 namespace LEGO.AsyncAPI.Writers
 {
-    using System.IO;
-
+    /// <summary>
+    /// Used to conver an AsyncApi schema into a yaml document.
+    /// </summary>
     public class AsyncApiYamlWriter : AsyncApiWriterBase
     {
+        private static readonly Regex YamlNumberRegex;
+        private static readonly char[] YamlIndicators;
+        private static readonly string[] YamlPlainStringForbiddenCobinations;
+        private static readonly string[] YamlPlainStringForbiddenTerminals;
+        private static readonly char[] YamlControlCharacters;
+
+        static AsyncApiYamlWriter()
```

Comment:
This was the custom functions from `SpecialCharacterStringExtensions` that just relate to yaml. I would like to delete this class but it has some remaining logic that should be moved to the JsonReader class. Just did not want to make even more edits in an already large review